omr/startup/omrvmstartup.cpp

```diff
@@ -463,6 +463,7 @@ OMR_Shutdown_VM(OMR_VM *omrVM, OMR_VMThread *omrVMThread)
 {
 	omr_error_t rc = OMR_ERROR_NONE;
 
+   // This path assumes that omrVM has been initialized
```

Comment:
I would move the check for omrVM up and wrap everything except the port library and thread library shutdown.